src/typing_extensions.py

```diff
@@ -1147,6 +1148,48 @@ def __repr__(self):
                                above.""")
 
 
+class _DefaultMixin:
+    """Mixin for TypeVarLike defaults."""
+
+    __slots__ = ()
+
+    def __init__(self, default):
+        if isinstance(default, (tuple, list)):
+            self.__default__ = tuple((typing._type_check(d, "Default must be a type")
+                                      for d in default))
+        elif default:
+            self.__default__ = typing._type_check(default, "Default must be a type")
+        else:
+            self.__default__ = None
+
+
+if sys.version_info >= (3, 12):
```

Comment:
We don't know yet that defaults will be in 3.12, and it definitely won't be in the first alpha (which I'm told is getting released this week). Let's unconditionally export our version of TypeVar/ParamSpec/TypeVarTuple even on 3.12.

Reply:
I agree, we can always push a new commit later once the PEP has been accepted. Pushed a new commit to revert these changes.